src/lab15.py

```diff
@@ -636,7 +636,7 @@ def layout(self):
         else:
             self.height = dpx(IFRAME_HEIGHT_PX + 2, self.zoom)
 
-        if self.node.frame:
+        if self.node.frame and self.node.frame.loaded:
```

Comment:
Ok, this is a good Q. Basically, in lab15 this doesn't do anything (if the frame isn't loaded, setting its frame height and width won't matter, because the frame isn't going to draw anything).

However, in lab16 the corresponding block of code also marks the root of the layout tree. We can't do that marking until the frame is loaded (because we need it to have a layout tree). So in lab16 we would need a change to the conditional or another `if` statement or something.

I changed the lab15 code because I figured there's no point in discussing this subtlety, but if you'd like we can only make the change in Chapter 16.